createAccount.php

```diff
@@ -0,0 +1,334 @@
+            );
+            
+            
+            
+            $errors = false;
+            if (!wereRequiredFieldsSubmitted($args, $required)) {
+                $errors = true;
+            }
+            $first = $args['first-name'];
+            $last = $args['last-name'];
+            $dateOfBirth = validateDate($args['birthdate']);
+            if (!$dateOfBirth) {
+                $errors = true;
+                echo 'bad dob';
+            }
+            $diagnosis_date=$args['diagnosis_date'];
+            /*$diagnosis_date = validateDate($args['diagnosis_date']);
+            if (!$diagnosis_date) {
+                $errors = true;
+                echo 'bad diagnosis date';
+            }*/
+
+            $hospital=$args['hospital'];
+
+            $cmethod=$args['cmethod'];
+            if(!$cmethod="call" and !$cmethod='text'){
+                $errors=true;
+                echo 'bad contact method';
+            }
+
+            $expected_treatment_end_date=$args['expected_treatment_end_date'];
+
+            $address = $args['address'];
+            $city = $args['city'];
+            $state = $args['state'];
+            if (!valueConstrainedTo($state, array('AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'FL', 'GA',
+                    'HI', 'IA', 'ID', 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD', 'ME',
+                    'MI', 'MN', 'MO', 'MS', 'MT', 'NC', 'ND', 'NE', 'NH', 'NJ', 'NM',
+                    'NV', 'NY', 'OH', 'OK', 'OR', 'PA', 'RI', 'SC', 'SD', 'TN', 'TX',
+                    'UT', 'VA', 'VT', 'WA', 'WI', 'WV', 'WY'))) {
+                $errors = true;
+            }
+            $zipcode = $args['zip'];
+            if (!validateZipcode($zipcode)) {
+                $errors = true;
+                echo 'bad zip';
+            }
+            $email = strtolower($args['email']);
+            $email = validateEmail($email);
+            if (!$email) {
+                $errors = true;
+                echo 'bad email';
+            }
+            $phone = validateAndFilterPhoneNumber($args['phone']);
+            if (!$phone) {
+                $errors = true;
+                echo 'bad phone';
+            }
+
+            $econtactName = $args['econtact-name'];
+            $agreement=$args['agreement'];
+            //requires signature to be same as contact parent, doesn't seem 
+            //like a guarentee so cut for now
+            /*if(validateAgreement($agreement, $econtactName)!=0){
+                $errors=true;
+                echo 'bad agreement';
+            }*/
+
+            /*$phone1 = validateAndFilterPhoneNumber($args['econtact-phone']);
+            if (!$econtactPhone) {
+                $errors = true;
+                echo 'bad e-contact phone';
+            }*/
+            
+            $diagnosis=$args['diagnosis'];
+            $permission_to_confirm=$args['permission_to_confirm'];
+           
+            /*
```

Comment:
Line 143 to 169, are all commented out is this needed?

Reply:
This code was for testing different ways to parse the checkbox question. Lines 159 to 166 are the base for the working code in the current iteration. The rest of the commented out code can be deleted.